[receiver/cloudfoundry] Enable component

crobert-1 opened the pull request:
**Description:** 
The Cloud Foundry receiver has already been implemented and merged into the contrib repository, however, testing was incomplete and the receiver wasn't enabled. This change adds the required unit testing and enables the receiver.

**Link to tracking Issue:**
#5320

**Testing:** 
This change is tests, and additional testing was done by running a Tanzu application service.

**Documentation:**
No documentation was added as this is a testing change.

crobert-1 committed:
Enable and add testing for PCF receiver

The PCF receiver has already been implemented and merged into the
contrib repository, however, testing was incomplete and the receiver
wasn't enabled by adding its factory to the valid receiver components
list. This change adds the required unit testing and enables the
receiver.

rmfitzpatrick commented on receiver/cloudfoundryreceiver/receiver_test.go:
```diff
@@ -14,4 +14,52 @@
+		nil,
+	)
+
+	require.Error(t, err)
```
could be helpful to have an `EqualError()` or similar to provide more info about the failure expectation

rmfitzpatrick commented on receiver/cloudfoundryreceiver/stream_test.go:
```diff
@@ -14,4 +14,100 @@
+		innerCtx,
+		INVALID_SHARD_ID)
+
+	require.Error(t, createErr)
```
same error content request

rmfitzpatrick commented on receiver/cloudfoundryreceiver/uaa_test.go:
```diff
@@ -14,4 +14,51 @@
+		cfg.UAA.Username,
+		cfg.UAA.Password)
+
+	require.Error(t, err)
```
same

crobert-1 committed:
Changes requested in PR

= Change require.Error to require.EqualError to specify expected
error.
- Resolve merge conflicts in dependency files.

jpkrohling commented on receiver/cloudfoundryreceiver/receiver_test.go:
```diff
@@ -14,4 +14,52 @@
+)
+
+// Test to make sure a new receiver can be created properly, started and shutdown with the default config
+func TestDefaultValidReceiver(t *testing.T) {
```
This should just be another test in this table test: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/013177e592b724dc217dc9ad958242db76158f4e/internal/components/receivers_test.go#L43

jpkrohling commented:
Please, also add an entry to the changelog.

crobert-1 committed:
Changes requested in PR

- Add tests to receiver test list
- Fix linting failures
- Add changes to changelog

crobert-1 committed:
Fix failing unit test

Unit tests are failing on github due to UAA auth errors. These
tests require a valid end point for authentication, but the github
environment does not meet this bar. These tests should be skipped
for now.

crobert-1 committed:
Fix linting error

- Linting error hit due to incorrect whitespace in test file
